library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import android.annotation.SuppressLint
+import android.os.Build
+import android.view.View
+import android.view.inspector.WindowInspector
+import java.lang.reflect.Field
+
+internal object WindowInspector {
+
+    fun getGlobalWindowViews(): List<View> {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            WindowInspector.getGlobalWindowViews()
+        } else {
+            getGlobalWindowViewApi29()
+        }
+    }
+
+    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @SuppressLint("PrivateApi", "DiscouragedPrivateApi")
+    private fun getGlobalWindowViewApi29(): List<View> {
+        try {
+            val globalWmClass = Class.forName("android.view.WindowManagerGlobal")
+            val globalWmInstance = globalWmClass.getMethod("getInstance").invoke(null)
+            val viewsField: Field = globalWmClass.getDeclaredField("mViews")
```

Comment:
It'd be better to have a lazy (static?) property to hold `globalWmClass` and `viewsField` as those will never change. That way you only compute them once